pkgs/by-name/li/libchewing/package.nix

```diff
@@ -4,32 +4,54 @@
   fetchFromGitHub,
   cmake,
   sqlite,
+  corrosion,
+  rustPlatform,
+  cargo,
+  rustc,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "libchewing";
-  version = "0.6.0";
+  version = "0.9.1";
 
   src = fetchFromGitHub {
     owner = "chewing";
     repo = "libchewing";
-    rev = "v${finalAttrs.version}";
-    sha256 = "sha256-X+4Rr5Mfc4qeJxmHczu4MKgHBvQN1rhqUrJSx8SFnDk=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-5aeAsvTiUMTm+ibNfJI57rzSUpJB7luhA/aWmTcnBj4=";
   };
 
-  buildInputs = [ sqlite ];
+  postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    substituteInPlace CMakeLists.txt \
+      --replace-fail "if(CMAKE_C_COMPILER_ID MATCHES GNU|^Clang)" "if((CMAKE_C_COMPILER_ID MATCHES GNU|^Clang) AND NOT APPLE)"
```

Comment:
Could you explain the substitution with a comment?

BTW, does it worth upstreaming?

Reply:
> I don't see what's wrong with this.

I mean, could you add a comment above this phase to explain why this line need patching, so that people reading this code in the future don't have to dig into the `libchewing` source to get what it means?